pkg/repository/local_git.go

```diff
@@ -0,0 +1,105 @@
+package repository
+
+import (
+	"context"
+	"errors"
+	"github.com/go-git/go-git/v5"
+	"os/exec"
+	"strings"
+)
+
+var (
+	errLocalGitCannotGetStatusDubiousOwner    = errors.New("detected dubious ownership in repository")
+	errLocalGitCannotGetStatusTooManyAttempts = errors.New("failed to get status: too many attempts")
+	errLocalGitCannotGetStatusCannotRecover   = errors.New("failed to get status: cannot recover")
+	errLocalGitInvalidStatusOutput            = errors.New("failed to get git status: unexpected status line")
+)
+
+type CommandExecutor interface {
+	RunCommand(ctx context.Context, dir string, name string, arg ...string) ([]byte, error)
+	LookPath(file string) (string, error)
+}
+
+type LocalExec struct{}
+
+func (*LocalExec) RunCommand(ctx context.Context, dir string, name string, arg ...string) ([]byte, error) {
+	c := exec.CommandContext(ctx, name, arg...)
+	c.Dir = dir
+	return c.Output()
+}
+
+func (*LocalExec) LookPath(file string) (string, error) {
+	return exec.LookPath(file)
+}
+
+type LocalGitInterface interface {
+	Status(ctx context.Context, dirPath string, fixAttempt int) (git.Status, error)
+	Exists() (string, error)
+	FixDubiousOwnershipConfig(path string) error
+	parseGitStatus(string) (git.Status, error)
+}
+
+type LocalGit struct {
+	gitPath string
+	exec    CommandExecutor
+}
+
+func NewLocalGit(gitPath string, exec CommandExecutor) *LocalGit {
+	return &LocalGit{
+		gitPath: gitPath,
+		exec:    exec,
+	}
+}
+
+func (lg *LocalGit) Status(ctx context.Context, dirPath string, fixAttempt int) (git.Status, error) {
```

Comment:
instead of run `git status` locally how about call graphql github API in order to get repo status and to not have two implementations depending of if user has `git` installed or not? It is feasible?